ipq40xx: Add kernel 4.19 support

robimarko opened the pull request:
This patch series introduces support for recently added 4.19 kernel to the ipq40xx target.
This reduces number of patches due to upstreamed patches in 4.19 kernel, while more stuff has been merged into 4.20 and backported from it.
Notable change is move from the staging MT29F driver to upstream SPI-NAND one.
Further details for all changes are in commit descriptions.

This series was tested on 8devices Jalapeno dev board and GL-Inet B1300.
So I invite anyone willing to test it on other ipq40xx to please do so and report their findings.

Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko committed:
ipq40xx: Add 4.19 kernel config

This adds the neccessary 4.19 kernel config.

Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko committed:
kernel: modules: usb: Add DWC3 Qualcomm kmod

Since kernel 4.18 support for Qualcomm glue layer was
moved from DWC3 OF Simple to a separate DWC3 QCOM module.
So lets add it and make it depend on 4.19 kernel and make
sure that DWC3 OF Simple is not included if 4.19 is used.

Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko committed:
ipq40xx: Use kmod-usb-dwc3-qcom by default

Since 4.18 we cant use DWC3 OF Simple anymore so we have to use
kmod-usb-dwc3-qcom.

Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko committed:
ipq40xx: Use upstream SPI-NAND driver instead of MT29F

Since 4.19 upstream kernel provides generic SPI-NAND
framework and vendor specific drivers.
Since only users of MT29F are 2 boards with Winbond
W25N01GV SPI-NAND for which support has been backported
from 4.20 we can drop the ever stuck in staging MT29F
driver and instead use the upstream driver.

Signed-off-by: Robert Marko <robimarko@gmail.com>

chunkeey commented:
hmpf, this is some bad timing as I pushed out a bigger ipq40xx related patches.

robimarko commented:
Ok, well I will rebase on current master.
Any reason why there is no "denx,fit" compatible on 8devices Jalapeno?
Also, why change the serial node name and add another patch when current nodes are in the upstream kernel?

robimarko committed:
ipq40xx: Adapt 4.19 DTS for upstream SPI-NAND driver

This adapts the necessary bindings for the upstream
SPI-NAND driver.

Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko commented:
PR has been rebased.
Aded patch was not needed as the changes are already in 4.19

chunkeey commented:
> Any reason why there is no "denx,fit" compatible on 8devices Jalapeno?

8devices Jalapeno uses UBI. The FIT (kernel + dtb + config), rootfs and rootfs_data are all in separate ubi volumes that don't mix well with MTD and fit-fw mtdsplitter. 

I didn't like the "denx,fit" compatible either so in my [V1 I opted for "openwrt,fit-firmware"](https://patchwork.ozlabs.org/patch/1010017/#2049763), but I was told to change it to ["denx,fit"](https://patchwork.ozlabs.org/patch/1010017/#2049222):

Also, I think the fix for the GL-B1300 will require another another rebase.

robimarko commented:
OK, thanks for the explanation.
Yeah, "denx,fit" is not really the most clear compatible.

Yeah, saw the mailing list and the B1300 issue with boot.
When its pushed to master I will rebase.

ynezz commented:
@robimarko  I've just tried to build  test this:

```
CONFIG_TARGET_ipq40xx=y
CONFIG_TARGET_MULTI_PROFILE=y
CONFIG_TARGET_ALL_PROFILES=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_8dev_jalapeno=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_asus_rt-ac58u=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_avm_fritzbox-4040=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_meraki_mr33=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_compex_wpj428=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_engenius_eap1300=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_glinet_gl-b1300=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_netgear_ex6100v2=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_netgear_ex6150v2=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_openmesh_a42=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_openmesh_a62=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_qcom_ap-dk01.1-c1=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_qcom_ap-dk04.1-c1=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_zyxel_nbg6617=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_zyxel_wre6606=y
CONFIG_ALL_KMODS=y
CONFIG_ALL_NONSHARED=y
```

```
a851d47 Merge branch 'PR1643'
5ae45dc ipq40xx: GL-B1300: convert to new partitions layout
b9501ce ath79: dts: Remove newly added default-state=off property
f54ac91 ar71xx/ath79: switch devices to the -ct driver and firmware
4472df2 bcm53xx: include WiFi firmware for devices with 4366C0
fa21162 linux-firmware: broadcom: package 4366C0 FullMAC firmware
8b4f6a1 linux-firmware: update to the commit from 2018-12-16
2574c86 kernel: backport ifconfig ioctl support for class e addresses
300ea20 ipq40xx: Adapt 4.19 DTS for upstream SPI-NAND driver
ee7fd26 ipq40xx: Use upstream SPI-NAND driver instead of MT29F
0965ae0 ipq40xx: Use kmod-usb-dwc3-qcom by default
a590b10 kernel: modules: usb: Add DWC3 Qualcomm kmod
89422ef ipq40xx: Add 4.19 kernel config
dd572ff ipq40xx: Add patches for 4.19
4effa2d ipq40xx: Add files for 4.19
```

```
diff --git a/target/linux/ipq40xx/Makefile b/target/linux/ipq40xx/Makefile
index 4ebaa91..e9cb44f 100644
--- a/target/linux/ipq40xx/Makefile
+++ b/target/linux/ipq40xx/Makefile
@@ -8,7 +8,7 @@ CPU_TYPE:=cortex-a7
 CPU_SUBTYPE:=neon-vfpv4
 MAINTAINER:=John Crispin <john@phrozen.org>
 
-KERNEL_PATCHVER:=4.14
+KERNEL_PATCHVER:=4.19
 
 KERNELNAME:=zImage Image dtbs
```

```
Package kmod-rtc-ds1307 is missing dependencies for the following libraries:
hwmon.ko
modules/other.mk:491: recipe for target '/home/petr/testing/openwrt/bin/targets/ipq40xx/generic/packages/kmod-rtc-ds1307_4.19.9-1_arm_cortex-a7_neon-vfpv4.ipk' failed
make[3]: *** [/home/petr/testing/openwrt/bin/targets/ipq40xx/generic/packages/kmod-rtc-ds1307_4.19.9-1_arm_cortex-a7_neon-vfpv4.ipk] Error 1
```

robimarko commented:
That is issue with the generic ds1307 module rather than with ipq40xx directly.
And I dont see how would RTC depend on hwmon?
There is no mention of it in the RTC Kconfig on kernel.org

ynezz commented:
> Hm, weirdly how it even passed kmod_all when generic 4.19 was added.

Because 4.19 is not build by default on the build bots yet? Anyway I've sent fixes for [this](http://lists.infradead.org/pipermail/openwrt-devel/2018-December/015209.html) and few other issues to the mailing list.

Now the build ends for me with following error:

```
WARNING: Image file ..linux-ipq40xx/netgear_ex6100v2-fit-uImage.itb is too big
Makefile:254: recipe for target '..openwrt-ipq40xx-netgear_ex6100v2-squashfs-factory.img' failed
```

BTW I've noticed that there is quite a lot of warnings in the DTS files:

```
image-qcom-ipq4018-jalapeno.dtb: Warning (unit_address_vs_reg): /memory: node has a reg or ranges property, but no unit name
image-qcom-ipq4018-jalapeno.dtb: Warning (avoid_unnecessary_addr_size): /soc/spi@78b5000/m25p80@0: unnecessary #address-cells/#size-cells without "ranges" or child "reg" property
image-qcom-ipq4018-jalapeno.dtb: Warning (avoid_unnecessary_addr_size): /soc/spi@78b5000/spi-nand@1: unnecessary #address-cells/#size-cells without "ranges" or child "reg" property
image-qcom-ipq4018-rt-ac58u.dtb: Warning (unit_address_vs_reg): /memory: node has a reg or ranges property, but no unit name
image-qcom-ipq4018-rt-ac58u.dtb: Warning (avoid_unnecessary_addr_size): /soc/spi@78b5000/flash@0: unnecessary #address-cells/#size-cells without "ranges" or child "reg" property
image-qcom-ipq4018-rt-ac58u.dtb: Warning (avoid_unnecessary_addr_size): /soc/spi@78b5000/spi-nand@1: unnecessary #address-cells/#size-cells without "ranges" or child "reg" property
image-qcom-ipq4018-fritz4040.dtb: Warning (unit_address_vs_reg): /memory: node has a reg or ranges property, but no unit name
image-qcom-ipq4018-fritz4040.dtb: Warning (avoid_unnecessary_addr_size): /soc/spi@78b5000/flash@0: unnecessary #address-cells/#size-cells without "ranges" or child "reg" property
image-qcom-ipq4028-wpj428.dtb: Warning (unit_address_vs_reg): /memory: node has a reg or ranges property, but no unit name
image-qcom-ipq4028-wpj428.dtb: Warning (avoid_unnecessary_addr_size): /soc/spi@78b5000/m25p80@0: unnecessary #address-cells/#size-cells without "ranges" or child "reg" property
image-qcom-ipq4018-eap1300.dtb: Warning (unit_address_vs_reg): /memory: node has a reg or ranges property, but no unit name
image-qcom-ipq4029-gl-b1300.dtb: Warning (unit_address_vs_reg): /memory: node has a reg or ranges property, but no unit name
image-qcom-ipq4029-mr33.dtb: Warning (unit_address_vs_reg): /memory: node has a reg or ranges property, but no unit name
image-qcom-ipq4018-ex6100v2.dtb: Warning (unit_address_vs_reg): /memory: node has a reg or ranges property, but no unit name
image-qcom-ipq4018-ex6150v2.dtb: Warning (unit_address_vs_reg): /memory: node has a reg or ranges property, but no unit name
image-qcom-ipq4018-a42.dtb: Warning (unit_address_vs_reg): /memory: node has a reg or ranges property, but no unit name
image-qcom-ipq4019-a62.dtb: Warning (unit_address_vs_reg): /memory: node has a reg or ranges property, but no unit name
image-qcom-ipq4019-ap.dk01.1-c1.dtb: Warning (unit_address_vs_reg): /memory: node has a reg or ranges property, but no unit name
image-qcom-ipq4019-ap.dk01.1-c1.dtb: Warning (unique_unit_address): /soc/spi@78b5000/mx25l25635f@0/kernel@180000: duplicate unit-address (also used in node /soc/spi@78b5000/mx25l25635f@0/firmware@180000)
image-qcom-ipq4019-ap.dk04.1-c1.dtb: Warning (unit_address_vs_reg): /memory: node has a reg or ranges property, but no unit name
image-qcom-ipq4018-nbg6617.dtb: Warning (unit_address_vs_reg): /memory: node has a reg or ranges property, but no unit name
image-qcom-ipq4018-nbg6617.dtb: Warning (avoid_unnecessary_addr_size): /soc/spi@78b5000/mx25l25635f@0: unnecessary #address-cells/#size-cells without "ranges" or child "reg" property
image-qcom-ipq4018-wre6606.dtb: Warning (unit_address_vs_reg): /memory: node has a reg or ranges property, but no unit name
8dev_jalapeno-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1: node has a unit name, but no reg property
8dev_jalapeno-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1/hash@1: node has a unit name, but no reg property
8dev_jalapeno-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1/hash@2: node has a unit name, but no reg property
8dev_jalapeno-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1: node has a unit name, but no reg property
8dev_jalapeno-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1/hash@1: node has a unit name, but no reg property
8dev_jalapeno-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1/hash@2: node has a unit name, but no reg property
8dev_jalapeno-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /configurations/config@1: node has a unit name, but no reg property
asus_rt-ac58u-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1: node has a unit name, but no reg property
asus_rt-ac58u-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1/hash@1: node has a unit name, but no reg property
asus_rt-ac58u-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1/hash@2: node has a unit name, but no reg property
asus_rt-ac58u-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1: node has a unit name, but no reg property
asus_rt-ac58u-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1/hash@1: node has a unit name, but no reg property
asus_rt-ac58u-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1/hash@2: node has a unit name, but no reg property
asus_rt-ac58u-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /configurations/config@1: node has a unit name, but no reg property
avm_fritzbox-4040-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1: node has a unit name, but no reg property
avm_fritzbox-4040-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1/hash@1: node has a unit name, but no reg property
avm_fritzbox-4040-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1/hash@2: node has a unit name, but no reg property
avm_fritzbox-4040-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1: node has a unit name, but no reg property
avm_fritzbox-4040-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1/hash@1: node has a unit name, but no reg property
avm_fritzbox-4040-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1/hash@2: node has a unit name, but no reg property
avm_fritzbox-4040-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /configurations/config@1: node has a unit name, but no reg property
compex_wpj428-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1: node has a unit name, but no reg property
compex_wpj428-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1/hash@1: node has a unit name, but no reg property
compex_wpj428-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1/hash@2: node has a unit name, but no reg property
compex_wpj428-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1: node has a unit name, but no reg property
compex_wpj428-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1/hash@1: node has a unit name, but no reg property
compex_wpj428-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1/hash@2: node has a unit name, but no reg property
compex_wpj428-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /configurations/config@4: node has a unit name, but no reg property
engenius_eap1300-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1: node has a unit name, but no reg property
engenius_eap1300-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1/hash@1: node has a unit name, but no reg property
engenius_eap1300-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1/hash@2: node has a unit name, but no reg property
engenius_eap1300-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1: node has a unit name, but no reg property
engenius_eap1300-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1/hash@1: node has a unit name, but no reg property
engenius_eap1300-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1/hash@2: node has a unit name, but no reg property
engenius_eap1300-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /configurations/config@4: node has a unit name, but no reg property
glinet_gl-b1300-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1: node has a unit name, but no reg property
glinet_gl-b1300-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1/hash@1: node has a unit name, but no reg property
glinet_gl-b1300-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1/hash@2: node has a unit name, but no reg property
glinet_gl-b1300-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1: node has a unit name, but no reg property
glinet_gl-b1300-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1/hash@1: node has a unit name, but no reg property
glinet_gl-b1300-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1/hash@2: node has a unit name, but no reg property
glinet_gl-b1300-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /configurations/config@1: node has a unit name, but no reg property
meraki_mr33-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1: node has a unit name, but no reg property
meraki_mr33-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1/hash@1: node has a unit name, but no reg property
meraki_mr33-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1/hash@2: node has a unit name, but no reg property
meraki_mr33-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1: node has a unit name, but no reg property
meraki_mr33-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1/hash@1: node has a unit name, but no reg property
meraki_mr33-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1/hash@2: node has a unit name, but no reg property
meraki_mr33-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /configurations/config@1: node has a unit name, but no reg property
netgear_ex6100v2-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1: node has a unit name, but no reg property
netgear_ex6100v2-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1/hash@1: node has a unit name, but no reg property
netgear_ex6100v2-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/kernel@1/hash@2: node has a unit name, but no reg property
netgear_ex6100v2-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1: node has a unit name, but no reg property
netgear_ex6100v2-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1/hash@1: node has a unit name, but no reg property
netgear_ex6100v2-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /images/fdt@1/hash@2: node has a unit name, but no reg property
netgear_ex6100v2-fit-uImage.itb.new.tmp: Warning (unit_address_vs_reg): /configurations/config@4: node has a unit name, but no reg property
```
Would be nice to get them fixed too :-)

robimarko commented:
Buildbots did not build it, but when generic 4.19 was added it should have already been fixed then as it affects all targets using 4.19 in the future.
I can confirm that your patch fixes ds1307.
I just tried building and openwrt-ipq40xx-netgear_ex6100v2-squashfs-factory.img is built fine, no warnings or errors.

Yeah, there are lot of warning thrown by DTC since the ipq40xx target was added.
I will take a look at them, but cant really promise anything

robimarko commented:
OK, So I rebased again and replaced the B1300 DTS with the fixed one from master.

ynezz commented:
> but when generic 4.19 was added it should have already been fixed

It's probably impossible, since it's generic and you don't know what kernel configs would look like in the platforms for 4.19, and it seems, that every platform triggers different issues, because of different patches etc.

> openwrt-ipq40xx-netgear_ex6100v2-squashfs-factory.img is built fine, no warnings or errors.

Interesting, care to share your build setup and steps? It fails here and @blocktrron has even [sent patch](http://lists.infradead.org/pipermail/openwrt-devel/2018-December/015216.html) to the mailing list to fix that build breakage

robimarko commented:
Well, my setup is rather simple Fedora29, simply selected the target with 4.19 set in Makefile and let it build images for all devices in the target.
Now, I am building with Target specfic packages and all kmods selected.
kmod-openvswitch is currently broken:
Package kmod-openvswitch is missing dependencies for the following libraries:
nf_conncount.ko
nsh.ko

Weirdly, kmods for it are pulled from the Packages repo along with the userspace apps.

ynezz commented:
@robimarko I can confirm, that his PR builds fine from scratch with my [upstream/fixes-for-419](https://github.com/ynezz/openwrt/tree/upstream/fixes-for-419) and with @blocktrron's [ipq40xx: clean up netgear-dni image generation](http://lists.infradead.org/pipermail/openwrt-devel/2018-December/015216.html) applied on top.

```
CONFIG_TARGET_ipq40xx=y
CONFIG_TARGET_MULTI_PROFILE=y
CONFIG_TARGET_ALL_PROFILES=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_8dev_jalapeno=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_asus_rt-ac58u=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_avm_fritzbox-4040=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_meraki_mr33=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_compex_wpj428=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_engenius_eap1300=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_glinet_gl-b1300=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_netgear_ex6100v2=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_netgear_ex6150v2=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_openmesh_a42=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_openmesh_a62=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_qcom_ap-dk01.1-c1=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_qcom_ap-dk04.1-c1=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_zyxel_nbg6617=y
CONFIG_TARGET_DEVICE_ipq40xx_DEVICE_zyxel_wre6606=y
CONFIG_ALL_KMODS=y
CONFIG_ALL_NONSHARED=y
```

```
f66c548 ipq40xx: clean up netgear-dni image generation
abaeeac Merge remote-tracking branch 'github-ynezz/upstream/fixes-for-419'
5d146e8 ipq40xx: Adapt 4.19 DTS for upstream SPI-NAND driver
8361d06 ipq40xx: Use upstream SPI-NAND driver instead of MT29F
b15ed6a ipq40xx: Use kmod-usb-dwc3-qcom by default
bb5b4ea kernel: modules: usb: Add DWC3 Qualcomm kmod
e310d76 ipq40xx: Add 4.19 kernel config
8d0d2d5 ipq40xx: Add patches for 4.19
a6fe7ca ipq40xx: Add files for 4.19
8cc0d0a kernel: Fix rtc-ds1307 dependency on hwmon-core for 4.19
0d160a2 kernel: Fix usb-chipidea dependency on ulpi.ko for 4.19
5318893 kernel: Fix kmod-w1 dependency on hwmon-core for 4.19
c040602 kernel: Add missing symbols to 4.19
09004e6 build: drop cameo-factory recipe
```

robimarko commented:
@ynezz Awesome, for me somehow NETGEAR EX61x0v2 images get built even without the @blocktrron  patch.
I am now cleaning up some DTC warnings.
I can fix those related to flash memory, but the other ones I cant really do anything as they are added by mkits.sh script.

robimarko committed:
ipq40xx: files-4.19: Clear some DTC warnings

DTC was throwing warnings like this:
image-qcom-ipq4018-jalapeno.dtb: Warning (avoid_unnecessary_addr_size): /soc/spi@78b5000/m25p80@0: unnecessary #address-cells/#size-cells without "ranges" or child "reg" property
image-qcom-ipq4018-jalapeno.dtb: Warning (avoid_unnecessary_addr_size): /soc/spi@78b5000/spi-nand@1: unnecessary #address-cells/#size-cells without "ranges" or child "reg" property

So lets fix this for our downstream boards.

Signed-off-by: Robert Marko <robimarko@gmail.com>

ynezz commented:
> I am now cleaning up some DTC warnings.

Nice!

> but the other ones I cant really do anything as they are added by mkits.sh script

Ok, so what about something like [this](https://github.com/ynezz/openwrt/commit/b3ec6e296b59df42503bcc3149ed2f3ae2b45584)? Just build tested it as I don't have any ipq40xx board. Maybe we could get rid of those remaining `/configurations/config@4` and similar warnings which are declared via `DEVICE_DTS_CONFIG` make variables, but I'm not sure how they're referenced and used, but maybe @chunkeey might know better.

> Also, the warning about /memory is about skeleton.dtsi and I dont want to carry a downstream patch just for that since upstream has not changed it.

Well, fixing it upstream seems to me like a proper solution.

robimarko commented:
That actually looks similar to what OpenEmbedded did.
We cant get rid of config@1 or any other config@whatever since those are usually hardcoded into bootloader and if its not there it wont boot.

As far as memory issue, it looks simple.
We can simply drop skeleton.dtsi and declare:
#address-cells = <1>;
#size-cells = <1>;

In the ipq4019.dtsi since even in skeleton.dtsi it states that it should not be used and will be dropped after all of its users stop using it.

ynezz commented:
> We cant get rid of config@1 or any other config@whatever since those are usually hardcoded into bootloader and if its not there it wont boot.

Maybe it would be better to just silence this warnings.

chunkeey commented:
> > We cant get rid of config@1 or any other config@whatever since those are usually hardcoded into bootloader and if its not there it wont boot.
> 
> Maybe it would be better to just silence this warnings.

That's a good question. I haven't tried myself. But it could be possible to just add the appropriate 

     #address-cells = <1>;
    #size-cells = <0>;

    and reg = <0>; reg = <1>; properties

i.e.:

          fdt {
 			#address-cells = <1>;
 			#size-cells = <0>;
 			description = \"${ARCH_UPPER} OpenWrt ${DEVICE} device tree blob\";
 			data = /incbin/(\"${DTB}\");
 			type = \"flat_dt\";
 			arch = \"${ARCH}\";
 			compression = \"none\";
 			hash@1 {
 				reg = <1>;
 				algo = \"crc32\";
 			};
 			hash@2 {
                                reg = <1>;
 				algo = \"sha1\";
 			};
 		};

and see if it still boots. (Of course, it doesn't solve all problems. The OM A42 with it's "config@om.a42" will be a pain.


As for the "memory node needs a unit-address". It's because of this include in upstream:
https://github.com/torvalds/linux/blob/master/arch/arm/boot/dts/qcom-ipq4019.dtsi#L16 

And judging from my interaction with linux-msm-soc, it's probably going to stay there :disappointed: .

robimarko commented:
@chunkeey I sent a patch upstream to remove skeleton.dtsi and simply add address-cells and size-cells to dts couple of hourse ago.
We will see how it goes.

chunkeey commented:
> @chunkeey I sent a patch upstream to remove skeleton.dtsi and simply add address-cells and size-cells to dts couple of hourse ago.
>We will see how it goes.

Certainly, It will be interesting to see what happens, maybe they don't notice what it means
and merge it. Because yeah, we can remove the property from the .dts that gets compiled.
But as said it yourself:

> Memory node is anyway filled by bootloader and does not need to be declared.

So, it gets added back by the uboot in the dtb the kernel sees. This is "FUN".

robimarko commented:
In practical terms it means nothing and a lot of ARM DTS has been converted to drop skeleton.dtsi, especially Armada.
Even in skeleton.dtsi it states the following since eventually everybody figured out that despite it looking as a good idea to have a minimal set of nodes included by default it became a pain due to DTS specs changing and anyway most of it was always getting rewritten.
* This file is deprecated, and will be removed once existing users have been
 * updated. New dts{,i} files should *not* include skeleton.dtsi, and should
 * instead explicitly provide the below nodes only as required.
 *
 * Skeleton device tree; the bare minimum needed to boot; just include and
 * add a compatible value.  The bootloader will typically populate the memory
 * node.

I built a image with the patch applied and it booted fine on 8devices Jalapeno, but I forgot to check whether RAM size is properly detected.
Will do that tommorow, gotta see how I can dump the dtb passed by bootloader and see if it injects the memory node.

@chunkeey  Also, I caught a weird thing today, in the upstream ipq4019.dtsi there is a RNG node defined but it seems to be not enabled by any device supported by OpenWrt.
Any specific reason for that or copy/pasting DTS simply missed it?

robimarko commented:
Just checked and there is only memory node, so its the bootloader that is patching that.
I can confirm that U-boot on Jalapeno does the same as MR33 .

Just checked and it was you who got the RNG node merged upstream and qcom-ipq4019-ap.dk01.1.dtsi has it enabled so it most likely somehow got dropped.
Will make a commit to re-enable it as HW PRNG is usefull.

robimarko commented:
@chunkeey It looks like HWPRNG is not working properly.
I have enabled the node and added selected CRYPTO_DEV_QCOM_RNG driver in kernel_menuconfig.
But rngtest is not happy:
root@OpenWrt:/dev# rngtest -c 1000 </dev/hwrng 
rngtest 6.6
Copyright (c) 2004 by Henrique de Moraes Holschuh
This is free software; see the source for copying conditions.  There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

rngtest: starting FIPS tests...
rngtest: error reading input: No such device

root@OpenWrt:/etc# rngd --rng-device=/dev/hwrng

Initalizing available sources

Failed to init entropy source hwrng

can't open any entropy source
Maybe RNG device modules are not loaded

chunkeey commented:
> @chunkeey It looks like HWPRNG is not working properly.
> I have enabled the node and added selected CRYPTO_DEV_QCOM_RNG driver in kernel_menuconfig.
> But rngtest is not happy:
> root@OpenWrt:/dev# rngtest -c 1000 </dev/hwrng
> rngtest 6.6
> Copyright (c) 2004 by Henrique de Moraes Holschuh
> This is free software; see the source for copying conditions. There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
> 
> rngtest: starting FIPS tests...
> rngtest: error reading input: No such device
> 
> root@OpenWrt:/etc# rngd --rng-device=/dev/hwrng
> 
> Initalizing available sources
> 
> Failed to init entropy source hwrng
> 
> can't open any entropy source
> Maybe RNG device modules are not loaded

Yeah, this is expected though. The prng does not have the character interface we all know and love/hate. If you want to do something useful with the pseudo-rng you have to utilize kcapi-rng.

robimarko commented:
Ahh, basically rng-tools do not work with it anymore.
Under 4.14 and with old msm-rng driver it works fine with rng-tools.

Still makes sense to enable it as it is built into SoC and has a proper driver.

robimarko committed:
ipq40xx: 4.19: Enable pseudo random number generator

IPQ40xx series has a HW pseudo random number generator built in.
It already has a node in the upstream ipq4019.dtsi so we just need to enable it.
Its driver has been rewritten to use crypto API so we dont have char interface like under 4.14 kernel.

Signed-off-by: Robert Marko <robimarko@gmail.com>

chunkeey commented:
@robimarko 
since you are looking through the DTS and collecting the patches, I too found another small bug:

https://git.openwrt.org/?p=openwrt/staging/chunkeey.git;a=commit;h=a4443e119285a0700ff6bbb7fc27cc29d0d37339

Other than that, I'm going to break out the MR33 (because of its "funny" AR8035) over the weekend and see what it does.

robimarko commented:
@chunkeey Once you push that patch into master I will rebase on it.
Please test the MR33 since there was a lot of manual rebasing of the patch due to timer changes in 4.15 which made quilt to fail to refresh automaticaly so I could have dropped something in the process.

robimarko committed:
ipq40xx: Add files for 4.19

This copies over files from 4.14

Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko committed:
ipq40xx: Add patches for 4.19

This adds the necessary patches for 4.19 kernel.
Upstreamed patches were dropped, backported upstreamed patches
from 4.20.
Drop Winbond ID patch since that NAND IC was upstreamed to use
SPI-NAND framework and support for it was backported from 4.20.
Rework ESSEDMA patches to compile under 4.19 due to timer changes,
Clément Péron did the hard work and his changes were taken from the
initial 4.19 PR.
MR33 changes had to be manually refreshed to apply.
Refresh other patches to apply.

Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko commented:
Rebased on master and applied your PCI fix to 4.19 DTS

chunkeey commented:
Meraki MR33 :disappointed: . Does anyone else has a device with proper NAND? Because it looks like the nand (not spi-nand) driver crashes:


```
[    0.932016] 12 fixed-partitions partitions found on MTD device qcom_nand.0
[    0.939306] Creating 12 MTD partitions on "qcom_nand.0":
[    0.946186] 0x000000000000-0x000000100000 : "sbl1"
[    0.951790] Unable to handle kernel NULL pointer dereference at virtual address 00000000
[    0.956237] pgd = (ptrval)       
[    0.964496] [00000000] *pgd=00000000
[    0.967002] Internal error: Oops: 80000005 [#1] SMP ARM
[    0.970726] Modules linked in:
[    0.975677] CPU: 0 PID: 1 Comm: swapper/0 Not tainted 4.19.9 #0
[    0.978801] Hardware name: Generic DT based system
[    0.984616] PC is at   (null)
[    0.989481] LR is at nand_block_isbad+0x90/0xa4
[    0.992515] pc : [<00000000>]    lr : [<c0592240>]    psr: 80000013
[    0.996856] sp : cf839d40  ip : 00000000  fp : cfae9e20
[    1.003105] r10: cf815810  r9 : 00000000  r8 : 00000000
[    1.008315] r7 : 00000000  r6 : 00000000  r5 : 00000001  r4 : cf815810
[    1.013527] r3 : 00000000  r2 : cfae9810  r1 : ffffffff  r0 : cf815810
[    1.020126] Flags: Nzcv  IRQs on  FIQs on  Mode SVC_32  ISA ARM  Segment none
[    1.026635] Control: 10c5387d  Table: 8020406a  DAC: 00000051
[    1.033837] Process swapper/0 (pid: 1, stack limit = 0x(ptrval))
[    1.039569] Stack: (0xcf839d40 to 0xcf83a000)
[    1.045648] 9d40: cf815810 cfae9c00 00000000 cfb43ac0 cfae9c00 00000000 00000000 c0580a94
[    1.049906] 9d60: 00100000 00000000 cfb43ac0 cf815810 00000000 00000000 c083c070 00020000
[    1.058067] 9d80: 000082dc 0000000c cfaed220 00000001 c0a182dc cf815810 0000000c 00000000
[    1.066224] 9da0: 00000000 c05811e4 00000000 00000000 cf815810 c0a182e4 c083be68 0000000c
[    1.074384] 9dc0: c42eee4c c0a1836c c0a182dc c0a58354 00000000 cf815810 cfdf5bd8 c0581164
[    1.082546] 9de0: 00000000 c0774eec c083c054 00000000 00000400 cfaed200 0000000c c0a1836c
[    1.090705] 9e00: 49464e4f cf815810 00000000 cf94d610 cfdf57a8 006080c0 cf94d610 cf995628
[    1.098863] 9e20: cfdf5a30 c057def4 cf995610 cf815810 cf94d610 c059d274 00000000 00000000
[    1.107024] 9e40: c080e268 c083ec98 cf94d600 cf94d610 cf949700 00000000 c0a195b8 cf94d610
[    1.115183] 9e60: c0a195b8 c0a581b8 00000000 00000000 c0a195b8 00000000 00000000 c0567f00
[    1.123343] 9e80: c0a581b4 cf94d610 c0a581b8 c0566768 cf94d610 c0a195b8 cf94d644 00000000
[    1.131502] 9ea0: 00000007 00000060 c0936e8c c0566b90 cf94d610 c0a195b8 c0566b00 c0564d0c
[    1.139664] 9ec0: cf807d58 cf947434 c0a195b8 cfa54d80 c0a177e0 c0565d04 c083edd4 c0915b90
[    1.147822] 9ee0: c0a195b8 c0a195b8 ffffe000 c0915ba0 00000000 c056726c c0a24200 ffffe000
[    1.155982] 9f00: c0915ba0 c030265c 00000003 00000000 00000000 c0349018 00000000 cf839f24
[    1.164141] 9f20: 00000000 00000000 c0886394 c084ae64 00000006 00000006 c08111dc 00000000
[    1.172301] 9f40: c0816f04 c0811250 00000000 cccccccd 00000000 c0a24200 c0a24200 c092c834
[    1.180460] 9f60: c092c83c 00000007 00000060 c0936e8c 00000000 c0900e20 00000006 00000006
[    1.188621] 9f80: 00000000 c09005a0 00000000 c0748d0c 00000000 00000000 00000000 00000000
[    1.196780] 9fa0: 00000000 c0748d14 00000000 c03010e8 00000000 00000000 00000000 00000000
[    1.204939] 9fc0: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
[    1.213099] 9fe0: 00000000 00000000 00000000 00000000 00000013 00000000 00000000 00000000
[    1.221275] [<c0592240>] (nand_block_isbad) from [<c0580a94>] (allocate_partition+0x7a0/0x7dc)
[    1.229424] [<c0580a94>] (allocate_partition) from [<c05811e4>] (add_mtd_partitions+0x58/0x10c)
[    1.237929] [<c05811e4>] (add_mtd_partitions) from [<c0581164>] (parse_mtd_partitions+0x310/0x338)
[    1.246524] [<c0581164>] (parse_mtd_partitions) from [<c057def4>] (mtd_device_parse_register+0x60/0x15c)
[    1.255555] [<c057def4>] (mtd_device_parse_register) from [<c059d274>] (qcom_nandc_probe+0x770/0x8f4)
[    1.265192] [<c059d274>] (qcom_nandc_probe) from [<c0567f00>] (platform_drv_probe+0x34/0x70)
```

chunkeey commented:
```
[  400.261142] ------------[ cut here ]------------
[  400.261203] WARNING: CPU: 2 PID: 1996 at drivers/irqchip/irq-gic.c:1016 gic_irq_domain_translate+0x74/0xf4
[  400.264841] Modules linked in: dwc3_qcom(+) dwc3 pppoe ppp_async iptable_nat ipt_MASQUERADE ath10k_pci ath10k_core ath xt_state xt_nat xt_conntrack xt_REDIRECT xt_FLOWOFFLOAD $
[  400.324670]  ip_tables hwmon crc_ccitt compat ledtrig_usbport xt_set ip_set_list_set ip_set_hash_netiface ip_set_hash_netport ip_set_hash_netnet ip_set_hash_net ip_set_hash_ne$
[  400.373367] CPU: 2 PID: 1996 Comm: kmodloader Not tainted 4.19.9 #0
[  400.394698] Hardware name: Generic DT based system
[  400.400791] [<c030e5b0>] (unwind_backtrace) from [<c030b074>] (show_stack+0x10/0x14)
[  400.405647] [<c030b074>] (show_stack) from [<c07355d0>] (dump_stack+0x88/0x9c)
[  400.413795] [<c07355d0>] (dump_stack) from [<c0317798>] (__warn+0xd4/0xf0)
[  400.421948] [<c0317798>] (__warn) from [<c0317884>] (warn_slowpath_null+0x40/0x48)
[  400.427688] [<c0317884>] (warn_slowpath_null) from [<c04f9cd0>] (gic_irq_domain_translate+0x74/0xf4)
[  400.435033] [<c04f9cd0>] (gic_irq_domain_translate) from [<c035af30>] (irq_create_fwspec_mapping+0xc4/0x328)
[  400.444311] [<c035af30>] (irq_create_fwspec_mapping) from [<c035b1e0>] (irq_create_of_mapping+0x4c/0x54)
[  400.454117] [<c035b1e0>] (irq_create_of_mapping) from [<c0614eec>] (of_irq_get+0x5c/0x94)
[  400.464509] [<c0614eec>] (of_irq_get) from [<c0614f3c>] (of_irq_to_resource+0x18/0xb0)
[  400.471807] [<c0614f3c>] (of_irq_to_resource) from [<c0614ff0>] (of_irq_to_resource_table+0x1c/0x48)
[  400.479656] [<c0614ff0>] (of_irq_to_resource_table) from [<c0610e08>] (of_device_alloc+0xfc/0x14c)
[  400.488968] [<c0610e08>] (of_device_alloc) from [<c0610ea0>] (of_platform_device_create_pdata+0x48/0xb8)
[  400.498574] [<c0610ea0>] (of_platform_device_create_pdata) from [<c061120c>] (of_platform_bus_create+0x2b8/0x340)
[  400.507365] [<c061120c>] (of_platform_bus_create) from [<c06113c4>] (of_platform_populate+0x54/0xa0)
[  400.517603] [<c06113c4>] (of_platform_populate) from [<bf4c06b4>] (dwc3_qcom_probe+0x538/0x76c [dwc3_qcom])
[  400.526551] [<bf4c06b4>] (dwc3_qcom_probe [dwc3_qcom]) from [<c0567f00>] (platform_drv_probe+0x34/0x70)
[  400.535952] [<c0567f00>] (platform_drv_probe) from [<c0566768>] (really_probe+0x14c/0x2b4)
[  400.545321] [<c0566768>] (really_probe) from [<c0566b90>] (__driver_attach+0x90/0xc4)
[  400.553855] [<c0566b90>] (__driver_attach) from [<c0564d0c>] (bus_for_each_dev+0x48/0x98)
[  400.561754] [<c0564d0c>] (bus_for_each_dev) from [<c0565d04>] (bus_add_driver+0xe8/0x200)
[  400.569934] [<c0565d04>] (bus_add_driver) from [<c056726c>] (driver_register+0xbc/0xf8)
[  400.578086] [<c056726c>] (driver_register) from [<c030265c>] (do_one_initcall+0x6c/0x160)
[  400.586173] [<c030265c>] (do_one_initcall) from [<c0380ea8>] (do_init_module+0x60/0x1dc)
[  400.594239] [<c0380ea8>] (do_init_module) from [<c03802ac>] (load_module+0x1d18/0x2060)
[  400.602402] [<c03802ac>] (load_module) from [<c0380724>] (sys_init_module+0x130/0x168)
[  400.610106] [<c0380724>] (sys_init_module) from [<c0301000>] (ret_fast_syscall+0x0/0x54)
[  400.617880] Exception stack(0xc69c3fa8 to 0xc69c3ff0)
[  400.626129] 3fa0:                   00000000 00000000 012b96e0 000025a0 00488c05 0000000e
[  400.631081] 3fc0: 00000000 00000000 00000005 00000080 000025a0 00000000 00488c05 00000000
[  400.639236] 3fe0: bebe7c1c bebe7c00 00487fe0 b6ef6538
[  400.649266] ---[ end trace 7bf942eb92f785db ]---
[  400.653380] ------------[ cut here ]------------
```

This is a warning triggered by broken DTs. I fixed that one locally

chunkeey commented:
> What exactly was broken?
I suppose the USB nodes

no longer... It was the NONE interrupt level and I found a few more things.

I added a note to tell people to add the dwc3-qcom module manually to their configuration. Unless you know a better way to automatically add the module as otherwise people compiling their own images with an existing OpenWrt .config will suddenly loose their USB functionality... and wonder why?

robimarko commented:
> > What exactly was broken?
> > I suppose the USB nodes
> 
> no longer... It was the NONE interrupt level and I found a few more things.
Awesome, looking forward to your patches.
> 
> I added a note to tell people to add the dwc3-qcom module manually to their configuration. Unless you know a better way to automatically add the module as otherwise people compiling their own images with an existing OpenWrt .config will suddenly loose their USB functionality... and wonder why?

Well, as far as I know, there is no way if you are using a old .config since it won't resolve dependencies again.
Although you should really not switch kernel versions using the old .config.

chunkeey commented:
> Well, as far as I know, there is no way if you are using a old .config since it won't resolve dependencies again.
Although you should really not switch kernel versions using the old .config.

True. It would be great if the kconfig logic could be like a package manager and have a way of automatically select the new flags, unless they have been forced disabled by the user.

By the way, do you know what happened to the usbport led-trigger in 4.19? It doesn't look it is working at all.

robimarko commented:
> > Well, as far as I know, there is no way if you are using a old .config since it won't resolve dependencies again.
> > Although you should really not switch kernel versions using the old .config.
> 
> True. It would be great if the kconfig logic could be like a package manager and have a way of automatically select the new flags, unless they have been forced disabled by the user.
> 
> By the way, do you know what happened to the usbport led-trigger in 4.19? It doesn't look it is working at all.

Hm, not really since my IPQ40xx devices dont have USB LED at all.
I may have a ath79 device with USB LED so I can check it out.
Is OpenWrt using the upstream CONFIG_USB_LED_TRIG or some downstream implementation?

chunkeey commented:
>Hm, not really since my IPQ40xx devices dont have USB LED at all.
I may have a ath79 device with USB LED so I can check it out.
Is OpenWrt using the upstream CONFIG_USB_LED_TRIG or some downstream implementation?

It's all upstream. @rmilecki did a wonderful job with the led trigger driver.

The bug can also be reproduced with 4.20 on my desktop. If I pick any led under /sys/class/leds (in my case, I'm using input2::scrolllock since I'm using an old PS/2 keyboard) and select "usbport" as the trigger. The "ports" directory will not contain any entries. However, if I reload the xhci_* module then suddenly all the: "usb2-port1  usb2-port10  usb2-port11  usb2-port12  usb2-port13 ..." entries appear.



> Also, can you post your DT fixes so I can test them out on my boards?

You know how it is :laughing:

robimarko commented:
@chunkeey Awesome for catching the LED trigger bug.
Should I rebase this PR on master?
I think that we are close to merging once the USB warning is sorted out.

chunkeey commented:
> @chunkeey Awesome for catching the LED trigger bug.

Time will tell... It's more of a hack for now and everybody upstream is busy with the holidays. Why not you too? :santa: In fact, I wanted to take a moment to look at the ipqess and qca8k patch John did a while ago. But this PR is eating up all the time.

> Should I rebase this PR on master?
I think that we are close to merging once the USB warning is sorted out.

Nah, the other USB warning was easy to fix. no need to rebase. Since KERNEL_PATCHVER is still at 4.14 missing patches can be added at a later time. 

Anyway: Merged. Thanks.

robimarko commented:
> > @chunkeey Awesome for catching the LED trigger bug.
> 
> Time will tell... It's more of a hack for now and everybody upstream is busy with the holidays. Why not you too? 🎅 In fact, I wanted to take a moment to look at the ipqess and qca8k patch John did a while ago. But this PR is eating up all the time.
> 
This is my idea of a vacation from work and university.
I am too hoping for a qca8k support at one point as essedma is working but is missing proper VLAN support instead of hardcoding them in DT.
> > Should I rebase this PR on master?
> > I think that we are close to merging once the USB warning is sorted out.
> 
> Nah, the other USB warning was easy to fix. no need to rebase. Since KERNEL_PATCHVER is still at 4.14 missing patches can be added at a later time.
> 
> Anyway: Merged. Thanks.

Thanks

robimarko commented:
@chunkeey Received no reply to the killing of skeleton.dtsi patch, took a quick look looking for other changes today and I can't believe that it actually got merged.
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/arch/arm/boot/dts/qcom-ipq4019.dtsi?h=v5.1-rc1&id=40122db87778970c96062c9f49b42cdf6809bf94

chunkeey commented:
Heh, this is going to be good. Now people will get confused about u-boots (QSDK reference u-boot does it in fdt_fixup_memory_banks() ) adding a /memory nodes so. So, who's going to be the poor soul that comes up with a good explanation for "hidden node" problem :question: 

Still :+1:  for following through!

robimarko commented:
Well, U-boot for pretty much all IPQ40XX or IPQ806X devices has been adding /memory and other nodes pretty much anyways.
So, this only makes warning during building go away.
I will most likely backport this as they are only warnings left on 4.19

chunkeey commented:
> Well, U-boot for pretty much all IPQ40XX or IPQ806X devices has been adding /memory and other nodes pretty much anyways.
So, this only makes warning during building go away.
I will most likely backport this as they are only warnings left on 4.19

Oh, well, these warnings are no more since cbbef976e2b358fa3039b02ada7e7904a9603653 I think. At least none of these show up in the lastest "Images" build step of ipq40xx:
http://phase1.builds.lede-project.org/builders/ipq40xx%2Fgeneric/builds/474/steps/images/logs/stdio

Also, in case of "/memory" it was a false positive from the beginning. Here: I've the full story straight from the horses mouth: https://www.spinics.net/lists/arm-kernel/msg494038.html
```
This has come up before and been beaten to death. Bottom line is plain "memory" is
allowed, and I plan to add that exception to dtc.
```

so, :man_shrugging: still nice that the "kill skeleton.dtsi" got accepted.

robimarko commented:
Well, that makes sense. I havent been checking for warnings lately.
I know its false positive, but still does not make sense to add the same
node multiple times.
I dont know how it got merged since its still showing new on Patchwork and
I havent received any update, accidentally saw it today.
PCI-E IRQ patch was also merged to 5.1.
I think that there are couple of patches in IPQ40xx that indicate wrong
version in which they were upstreamed, will look at it and push patch to
change names if that is the case.

On Wed, 20 Mar 2019 at 17:15, Christian Lamparter <notifications@github.com>
wrote:

> Well, U-boot for pretty much all IPQ40XX or IPQ806X devices has been
> adding /memory and other nodes pretty much anyways.
> So, this only makes warning during building go away.
> I will most likely backport this as they are only warnings left on 4.19
>
> Oh, well, these warnings are no more since cbbef97
> <https://github.com/openwrt/openwrt/commit/cbbef976e2b358fa3039b02ada7e7904a9603653>
> I think. At least none of these show up in the lastest "Images" build step
> of ipq40xx:
>
> http://phase1.builds.lede-project.org/builders/ipq40xx%2Fgeneric/builds/474/steps/images/logs/stdio
>
> Also, in case of "/memory" it was a false positive from the beginning.
> Here: I've the full story straight from the horses mouth:
> https://www.spinics.net/lists/arm-kernel/msg494038.html
>
> This has come up before and been beaten to death. Bottom line is plain "memory" is
>
> allowed, and I plan to add that exception to dtc.
>
>
> so, 🤷‍♂️ still nice that the "kill skeleton.dtsi" got accepted.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/1643#issuecomment-474908518>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABTDsFpm2bypF1CbT3HxCk41PzuJNX7aks5vYl6qgaJpZM4ZVel9>
> .
>
